services/core-api/app/api/now_applications/resources/now_application_progress_resource.py

```diff
@@ -45,6 +46,11 @@ def post(self, application_guid, application_progress_status_code):
                                                              progress_status_code)
             now_progress = new_now_progress
 
+        if identity.now_application is not None and identity.now_application.now_application_status_code != "REF" and progress_status_code == "REF" or progress_status_code == "CON" or progress_status_code == "PUB":
```

Comment:
A NoW can only have it's status changed to REF, CON, or PUB if it is not currently in REF?